src/components/Integrations/BigQuery/ControlPanel/InputSpanForm.js

```diff
@@ -0,0 +1,143 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { Formik, Field, Form, FieldArray } from 'formik';
+import { useIntl } from 'react-intl';
+import { validateEmail } from '../../../../validations';
+import { isUniqueArray } from '../../../../utils';
+import { Loading } from '../../../Loading/Loading';
+
+export const InputSpanForm = () => {
+  const [initialValues, setInitialValues] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const intl = useIntl();
+  const _ = useCallback(
```

Comment:
why use useCallback hook here?

Reply:
It is not necessary to use it but there are advantages to doing so

The useCallback hook has a primary and specific function: avoid unnecessary re-renders in your code, making your application faster and more efficient.

The useCallback hook receives a function as a parameter, and also an array of dependencies. The useCallback hook will return a memoized version of the callback, and it’ll only be changed if one of the dependencies has changed.

in this case it is only necessary for the function to reload the language files if they change, and not to load it into memory every time the _ is executed, that is why I think it is better to use the callback.

Reply:
Mmm, if I am not wrong, without memorizing this function there will not be more renderings. Because this function is not a dependency in anywhere.

Am I missing something?

Reply:
I'm not sure if this is the correct way to use useCallback. Also, I don't think it's very useful in this situation.
**React docs:** 
![image](https://user-images.githubusercontent.com/84402180/124660614-0e23e600-de6c-11eb-8803-8166d07c1675.png)

Reply:
ok, suggest accepted and I'm going to remove the callback

Reply:
Hecho, resuelvo la conversación...